### DISCUSSION TITLE
Interactive scripting

### DISCUSSION THREAD

**Discussion by chengchangwu** [pest-parser/pest#652] [General]
Using Python's interpreter I can input a part of the if clause, the interpreter will wait until I input all of the clause. How can I achieve this with pest?

```
>>> a = 33
>>> b = 200
>>> if b > a:
...   print("b is greater than a")
... 
b is greater than a
>>> 
```

**Comment by dragostis** [comment]
pest does not support streamed inputs for now, but you can consider having a grammar that parses input per lines.

**Comment by CAD97** [comment]
If your grammar requires a closing indicator, you can just attempt a parse on every newline and then figure out if it failed because of not being complete or not being correct.

It's not automatic and doesn't work for all grammars, but it should work.

**Comment by ice1000** [comment]
The IntelliJ plugin might be what you're looking for -- it is GUI, but you can edit grammar and the-file-parsed-by-grammar and real-time syntax highlight your file.
